torch/_dynamo/variables/misc.py

```diff
@@ -635,6 +635,47 @@ def call_method(
         return super().call_method(tx, name, args, kwargs)
 
 
+class GetAttrFunctionVariable(VariableTracker):
+    def __init__(self, get_fn, name, **kwargs):
+        super().__init__(**kwargs)
+        self.get_fn = get_fn
+        self.name = name
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        assert len(args) == 1 and len(kwargs) == 0
+        return args[0].var_getattr(tx, self.name)
+
+    def is_python_constant(self):
+        return True
+
+    def as_python_constant(self):
+        return self.get_fn
+
+
+class GetSetDescriptorVariable(VariableTracker):
+    def __init__(self, desc, **kwargs):
+        super().__init__(**kwargs)
+        self.desc = desc
+
+    def var_getattr(self, tx, name):
+        if name == "__get__":
+            from .builder import VariableBuilder
+
+            return VariableBuilder(tx, AttrSource(self.source, "__get__"))(
+                self.desc.__get__
+            )
+        else:
+            return super().var_getattr(tx, name)
+
+    def is_python_constant(self):
+        return True
+
+    def as_python_constant(self):
+        return self.desc
```

Comment:
Actually, why aren't these just both ConstantVariable 🤔

Reply:
Yeah let me try this, it should prob work. Originally I was thinking BaseUserFunctionVariable, but in the end only one var_getattr needs to be defined so ConstantVariable looks like it might be cleanest.